src/dimension.c

```diff
@@ -152,21 +152,47 @@ hyperspace_get_num_dimensions_by_type(Hyperspace *hs, DimensionType type)
 }
 
 static inline DimensionType
-dimension_type(TupleInfo *ti)
+get_dimension_type(char dimension_type)
 {
-	if (slot_attisnull(ti->slot, Anum_dimension_interval_length) &&
-		!slot_attisnull(ti->slot, Anum_dimension_num_slices))
-		return DIMENSION_TYPE_CLOSED;
-
-	if (!slot_attisnull(ti->slot, Anum_dimension_interval_length) &&
-		slot_attisnull(ti->slot, Anum_dimension_num_slices))
-		return DIMENSION_TYPE_OPEN;
+	switch (dimension_type)
+	{
+		case FD_DIMENSION_OPEN:
+			return DIMENSION_TYPE_OPEN;
+		case FD_DIMENSION_CLOSED:
+			return DIMENSION_TYPE_CLOSED;
+		case FD_DIMENSION_CORRELATED:
+			return DIMENSION_TYPE_CORRELATED;
+		default:
+			/* should not be possible */
+			ereport(ERROR, (errmsg("unexpected dimension type %c", dimension_type)));
+			break;
+	}
 
-	elog(ERROR, "invalid partitioning dimension");
 	/* suppress compiler warning on MSVC */
 	return DIMENSION_TYPE_ANY;
 }
 
+static inline char
+get_dimension_char_type(DimensionType dimension_type)
```

Comment:
I think it is possible to get rid of this translation between char and enum per my suggestion above.

Reply:
Done.